crates/winch/src/compiler.rs

```diff
@@ -92,8 +92,22 @@ impl wasmtime_environ::Compiler for Compiler {
         types: &ModuleTypes,
         index: DefinedFuncIndex,
     ) -> Result<Box<dyn Any + Send>, CompileError> {
-        let _ = (translation, types, index);
-        todo!()
+        let index = translation.module.func_index(index);
+        let sig = translation.module.functions[index].signature;
+        let ty = &types[sig];
+        let wasm_ty = wasmparser::FuncType::new(
+            ty.params().iter().copied().map(Into::into),
+            ty.returns().iter().copied().map(Into::into),
+        );
+        let env = FuncEnv::new(&translation.module, translation.get_types(), &self.isa);
+        let buffer = self
+            .isa
+            .compile_trampoline(&wasm_ty, index.as_u32(), &env, TrampolineKind::ArrayToWasm)
```

Comment:
Not 100% related to this PR per se, but I've found that using all the `FooIndex` types has been quite helpful throughout Cranelift and the existing wasm lowering to ensure that everything is kept straight. For example I saw this and immediately thought "wait that's bad because this is a defined index" and then looked around a bit before I noticed above that the variable was shadowed with the `FuncIndex`. This `index.as_u32()` is fine as-is, but it might be useful to use the `wasmtime-types::*` index types throughout winch in the future (only if you agree of course!)